python/paddle/fluid/tests/unittests/test_while_loop_op.py

```diff
@@ -189,63 +175,84 @@ def body(i, x):
-            i = layers.increment(i)
-            return [i, x, y]
+class TestApiWhileLoop_NestedWithBackwardAndLoDTensorArray(unittest.TestCase):
+    def test_nested_net_with_backward_and_lodtensor(self):
+        def external_cond(i, j, x, mem_array):
+            return layers.less_than(i, array_len)
+
+        def external_body(i, j, x, mem_array):
+            def internal_cond(j, x, mem_array):
+                return layers.less_than(j, array_len2)
+
+            def internal_body(j, x, mem_array):
+                inner_data = layers.array_read(array=data_array, i=j)
+                inner_prev = layers.array_read(array=mem_array, i=j)
+                inner_sum_0 = layers.elementwise_add(x=inner_data, y=inner_prev)
+                inner_sum_1 = layers.elementwise_add(x=x, y=inner_sum_0)
+                j = layers.increment(x=j, in_place=True)
+                layers.array_write(inner_sum_1, i=j, array=mem_array)
+                return [j, x, mem_array]
+
+            outer_data = layers.array_read(array=data_array, i=i)
+            outer_prev = layers.array_read(array=mem_array, i=i)
+            outer_sum_0 = layers.elementwise_add(x=outer_data, y=outer_prev)
+            outer_sum_1 = layers.elementwise_add(x=x, y=outer_sum_0)
+            i = layers.increment(x=i, in_place=True)
+            layers.array_write(outer_sum_1, i=i, array=mem_array)
+            j, x, mem_array = layers.while_loop(internal_cond, internal_body,
+                                                [j, x, mem_array])
+            return [i, j, x, mem_array]
 
         main_program = Program()
         startup_program = Program()
-
         with fluid.program_guard(main_program, startup_program):
-            i = layers.data(
-                name='i', shape=[1], dtype='float32', append_batch_size=False)
-            i.stop_gradient = False
-            ten = layers.fill_constant(shape=[1], dtype='float32', value=10)
-            five = layers.fill_constant(shape=[1], dtype='float32', value=5)
-            x = layers.data(
-                name='x', shape=[1], dtype='float32', append_batch_size=False)
+            d0 = fluid.data(name='d0', shape=[10], dtype='float32')
+            d1 = fluid.data(name='d1', shape=[10], dtype='float32')
+            d2 = fluid.data(name='d2', shape=[10], dtype='float32')
+            x = fluid.data(name='x', shape=[10], dtype='float32')
             x.stop_gradient = False
-            y = layers.data(
-                name='y', shape=[1], dtype='float32', append_batch_size=False)
-            y.stop_gradient = False
-            out = layers.while_loop(external_cond, external_body, [i, x, y])
-
-            mean = layers.mean(out[2])
-            append_backward(mean)
-
-        place = fluid.CUDAPlace(0) if core.is_compiled_with_cuda(
-        ) else fluid.CPUPlace()
-        exe = fluid.Executor(place)
+            i = layers.zeros(shape=[1], dtype='int64')
+            i.stop_gradient = True
```

Comment:
You don't have to put unnecessary stop_gradient, since these are test code we would like this PR in release 1.7. I will approve this PR and please change them in a follow up PR.